poppy/poppy_core.py

```diff
@@ -199,7 +199,7 @@ def __mul__(self, optic):
 
     def __iadd__(self, wave):
         """Add another wavefront to this one"""
-        if not isinstance(wave, Wavefront):
+        if not isinstance(wave, BaseWavefront):
```

Comment:
Unrelated to the main point of this PR, but looking at this function makes me realize it needs a bit more careful handling. It should only be possible to add wavefronts of the same class together. And with the same pixel scale, and wavelength, and at the same plane type if Fraunhofer...